debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java

```diff
@@ -343,6 +345,15 @@ public class OracleConnectorConfig extends HistorizedRelationalDatabaseConnector
                     System.lineSeparator() +
                     "infinispan_remote - This option uses a remote Infinispan cluster to buffer transaction data and persist it to disk.");
 
+    public static final Field LOG_MINING_BUFFER_MEMORY_TRANSACTION_EVENTS_THRESHOLD = Field.create("log.mining.buffer.memory.transaction.events.threshold")
+            .withDisplayName("Transaction events threshold when use log.mining.buffer.type='memory'.")
+            .withType(Type.LONG)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.LOW)
+            .withDefault(DEFAULT_TRANSACTION_EVENTS_THRESHOLD)
+            .withDescription("The transaction events threshold when use log.mining.buffer.type='memory'. " +
+                    "If events count exceeds this threshold the transaction will be abandoned.");
+
```

Comment:
I don't think we necessarily need to limit this to just the memory buffer type.  There is no harm in this just being named `log.mining.buffer.transaction.events.threshold` because the default is `0` (there is no limit).  This would allow either implementation to take advantage of this if the user enabled it, giving the most flexibility.  You should be able to pull the logic up into the abstract class for both to use.  Don't forget to change variable names and java docs accordinglly.